contracts/contracts/buyback/AbstractBuyback.sol

```diff
@@ -40,286 +26,44 @@ abstract contract AbstractBuyback is Initializable, Strategizable {
-     */
-    function swapForOGN(
-        uint256 oTokenAmount,
-        uint256 minOGN,
-        bytes calldata swapData
-    ) external onlyGovernorOrStrategist nonReentrant {
-        (uint256 _amountForOGN, ) = _updateBuybackSplits();
-        require(_amountForOGN >= oTokenAmount, "Balance underflow");
-        require(rewardsSource != address(0), "RewardsSource contract not set");
-
-        unchecked {
-            // Subtract the amount to swap from net balance
-            balanceForOGN = _amountForOGN - oTokenAmount;
-        }
-
-        uint256 ognReceived = _swapToken(ogn, oTokenAmount, minOGN, swapData);
-
-        // Transfer OGN received to RewardsSource contract
-        // slither-disable-next-line unchecked-transfer unused-return
-        IERC20(ogn).transfer(rewardsSource, ognReceived);
-    }
-
-    /**
-     * @dev Swaps `oTokenAmount` to CVX
-     * @param oTokenAmount Amount of OUSD/OETH to swap
-     * @param minCVX Minimum CVX to receive for oTokenAmount
-     * @param swapData 1inch Swap Data
-     */
-    function swapForCVX(
-        uint256 oTokenAmount,
-        uint256 minCVX,
-        bytes calldata swapData
-    ) external onlyGovernorOrStrategist nonReentrant {
-        (, uint256 _amountForCVX) = _updateBuybackSplits();
-        require(_amountForCVX >= oTokenAmount, "Balance underflow");
-
-        unchecked {
-            // Subtract the amount to swap from net balance
-            balanceForCVX = _amountForCVX - oTokenAmount;
-        }
-
-        uint256 cvxReceived = _swapToken(cvx, oTokenAmount, minCVX, swapData);
-
-        // Lock all CVX
-        _lockAllCVX(cvxReceived);
-    }
-
-    /**
-     * @dev Locks all CVX held by the contract on behalf of the Treasury Manager
-     */
-    function lockAllCVX() external onlyGovernorOrStrategist {
-        _lockAllCVX(IERC20(cvx).balanceOf(address(this)));
-    }
-
-    function _lockAllCVX(uint256 cvxAmount) internal {
-        require(
-            treasuryManager != address(0),
-            "Treasury manager address not set"
-        );
-
-        // Lock all available CVX on behalf of `treasuryManager`
-        ICVXLocker(cvxLocker).lock(treasuryManager, cvxAmount, 0);
-    }
-
-    /**
-     * @dev Approve CVX Locker to move CVX held by this contract
-     */
-    function safeApproveAllTokens() external onlyGovernorOrStrategist {
-        IERC20(cvx).safeApprove(cvxLocker, type(uint256).max);
     }
 
     /**
      * @notice Owner function to withdraw a specific amount of a token
+     *          to the guardian (2/8 multisig)
      * @param token token to be transferered
      * @param amount amount of the token to be transferred
      */
     function transferToken(address token, uint256 amount)
         external
-        onlyGovernor
+        onlyGovernorOrStrategist
         nonReentrant
     {
-        IERC20(token).safeTransfer(_governor(), amount);
+        IERC20(token).safeTransfer(strategistAddr, amount);
```

Comment:
nit: a require checking that strategist `address != 0` might be nice. It is easier to forget setting the strategist, while the governor will probably always be set

Reply:
OUSD and OETH have a `_to != address(0)` check but yeah good to add a check here for all other tokens. Have added it: https://github.com/OriginProtocol/origin-dollar/pull/2506/commits/f3ec40c0312201e737742b60511038fc01777c3f